src/descriptor/checksum.rs

```diff
@@ -41,12 +41,21 @@ fn poly_mod(mut c: u64, val: u64) -> u64 {
     c
 }
 
-/// Computes the checksum bytes of a descriptor
-pub fn get_checksum_bytes(desc: &str) -> Result<[u8; 8], DescriptorError> {
+/// Computes the checksum bytes of a descriptor.
+/// `exclude_hash = true` ignores all data after the first '#' (inclusive).
+pub fn get_checksum_bytes(mut desc: &str, exclude_hash: bool) -> Result<[u8; 8], DescriptorError> {
```

Comment:
Instead of innocently naming it `exlcude_hash` can we name it something dangerous like `with_inception_bug`? That will clear the intent that this value should never be set to `false` unless wrong checksum calculation is intended.

In the current state always using this value as `true` works for both the cases of w/wo checksum in descriptor. So Ideally everywhere else we should just use `get_checksum` and avoid using this function. Better to remove from public interface also?

Reply:
This is though. I still think we need to keep `get_checksum_bytes` in the public interface, but I feel like the `exclude_hash` logic should be hidden (maybe in an internal `get_checksum_internal`?).

Also, I want everyone that updates to bdk 0.23 and used these methods to know that something has changed (optimally by displaying a warning or something when compiling), or otherwise I suppose some nasty bugs will arise. Same for `get_checksum`.

I honestly don't know how to handle this...

Reply:
@danielabrozzoni Maybe we can just rename the functions all together, this way no one gets confused.

- `calc_checksum_bytes_internal(desc: &str, exclude_hash: bool)`
- `calc_checksum_bytes(desc: &str) // avoids checksum like a boss`
- `calc_checksum(desc: &str) // also avoids checksum like a boss`

Reply:
ACK on the above.. Keeping the API same would be helpful.. The name changes looks good to me.. That with a good explainer would do..

Reply:
@evanlinjin looks like everyone is onboard with the above proposed naming, but if I understand correctly these changes are needed:
 - [ ] rename `get_checksum_bytes` to `calc_checksum_bytes_internal` and make not public
 - [ ] add deprecated public `calc_checksum_bytes` that calls `calc_checksum_bytes_internal` with exclude_hash == false, this replicates current behavior but warns current users they need to change to `calc_checksum`.
 - [ ] add `calc_checksum` that calls `calc_checksum_bytes_internal` with exclude_hash == true, this is new function with new name that users should use instead of depricated `calc_checksum_bytes`
 - [ ] in new PR for 0.25 release (asuming this PR goes into 0.24) consolidate correct behavior into `calc_checksum` and remove deprecated and internal functions.

We can discuss on the call Tues too.

Reply:
I'm assuming if we implement `calc_checksum_bytes_internal`, we wouldn't need `calc_checksum_internal`, but I might be wrong.

Reply:
I added a commit to:
1. make  `calc_checksum_bytes_internal` pub(crate)
2. add backward compatible but deprecated `get_checksum_bytes` and `get_checksum` functions
3. rename replacement functions `calc_checksum_bytes` and `calc_checksum`
4. added a simple test to prove the new functions work as expected with a descriptor that already includes a checksum

I think this covers all the bases. Let me know if it looks OK or if you see anything else I should do.

Reply:
Resolving this one as fixed in latest push..